test/testdrive/kafka-upsert-sources.td

```diff
@@ -625,3 +625,50 @@ $ kafka-ingest format=bytes key-format=bytes key-terminator=: topic=format-json-
 "\"int\"" 99
 "\"object\"" "{\"y\":\"z\"}"
 "\"str\"" "\"bye\""
+
+
+#
+# Test Inline error handling with value decode failures
+#
+
+$ kafka-create-topic topic=inline-value-errors-1 partitions=1
+
+$ kafka-ingest format=bytes key-format=bytes key-terminator=: topic=inline-value-errors-1
+val1:{"a":,"c":"d"}
+val2:[1,2,
+
+> CREATE CLUSTER inline_error_cluster SIZE '${arg.default-storage-size}';
+
+# This source will not inline errors and should propagate them
+> CREATE SOURCE value_decode_error
+  IN CLUSTER inline_error_cluster
+  FROM KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-inline-value-errors-1-${testdrive.seed}')
+  KEY FORMAT TEXT
+  VALUE FORMAT JSON
+  ENVELOPE UPSERT (VALUE DECODING ERRORS = (PROPAGATE));
+
+! SELECT * FROM value_decode_error
+contains:Envelope error: Upsert: Value error: Bytes: Failed to decode JSON:
+
+> DROP SOURCE value_decode_error
+
+# This source will inline errors only
+> CREATE SOURCE value_decode_error
+  IN CLUSTER inline_error_cluster
+  FROM KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-inline-value-errors-1-${testdrive.seed}')
+  KEY FORMAT TEXT
+  VALUE FORMAT JSON
+  ENVELOPE UPSERT (VALUE DECODING ERRORS = (INLINE));
+
+> SELECT key, value::text, error::text FROM value_decode_error
+val1 <null> "(\"Bytes: Failed to decode JSON: {\"\"a\"\":,\"\"c\"\":\"\"d\"\"} (original bytes: [7b, 22, 61, 22, 3a, 2c, 22, 63, 22, 3a, 22, 64, 22, 7d])\",)"
+val2 <null> "(\"Bytes: Failed to decode JSON: [1,2, (original bytes: [5b, 31, 2c, 32, 2c])\",)"
+
+# replace the bad values with good ones
+$ kafka-ingest format=bytes key-format=bytes key-terminator=: topic=inline-value-errors-1
+val1:{"a": 1,"c":"d"}
+val2:[1,2,3]
+
+> SELECT key, value::text, error::text FROM value_decode_error
+val1 "(\"{\"\"a\"\":1,\"\"c\"\":\"\"d\"\"}\")" <null>
+val2 "(\"[1,2,3]\")" <null>
```

Comment:
Seems worth testing `(INLINE, PROPAGATE)` behavior here too.

Reply:
I'll add a test for it - though I realized that it'd be impossible to validate the inline error gets written since `PROPAGATE` would mean that the source will error out when queried. Unless you know of some mechanism that would allow also validating the inline-error in the source data shard?

Reply:
> Unless you know of some mechanism that would allow also validating the inline-error in the source data shard?

Yeah, you can create a sink on the source. You should be able to see the data still flow through successfully to the sink even though the source will error out when queried.

Reply:
discussed on slack, but we are removing the `PROPAGATE` option such that this case no longer applies